libs/cypress-indexeddb/src/lib/cypress-indexeddb.ts

```diff
@@ -0,0 +1,167 @@
+  // -- This is a parent command --
+  Cypress.Commands.add('clearIndexedDb', (databaseName: string) => {
+    return cy.window().then(async (window: Cypress.AUTWindow) => {
+      const promise = new Promise<void>((resolve, reject) => {
+        const deleteDb = window.indexedDB.deleteDatabase(databaseName);
+        deleteDb.onsuccess = () => {
+          if (DATABASES.has(databaseName)) {
+            DATABASES.delete(databaseName);
+          }
+          resolve();
+        };
+        deleteDb.onerror = (e) => reject(e);
+      });
+      try {
+        await promise;
+        cy.log(`"${databaseName}" cleared`);
+      } catch (e) {
+        cy.log(`error during "${databaseName}" clear operation`);
+        cy.log(e);
+      }
+      return window;
+    });
+  });
+
+  Cypress.Commands.add(
+    'openIndexedDb',
+    (databaseName: string, versionConfiguredByUser?: number) => {
+      return cy.window().then(async (window: Cypress.AUTWindow) => {
+        const db = DATABASES.get(databaseName);
+        if (db) {
+          return Promise.resolve(db);
+        }
+        const newVersion = versionConfiguredByUser || 2;
+        const request = window.indexedDB.open(databaseName, newVersion);
+        return new Promise<IDBDatabase>((resolve, reject) => {
+          request.onerror = reject;
+          request.onupgradeneeded = (event: any) => {
+            const db = event.target?.result;
+            DATABASES.set(databaseName, db);
+            resolve(db);
+          };
+        });
+      });
+    }
+  );
+
+  Cypress.Commands.add(
+    `createObjectStore`,
+    { prevSubject: true },
+    (database: IDBDatabase, storeName: string) => {
+      if (database?.constructor?.name !== 'IDBDatabase') {
+        throw new Error(
+          `You tried to use the 'getObjectStore' method without calling 'openIndexedDb' first`
+        );
+      }
+      return new Promise((resolve, reject) => {
+        try {
+          let store: IDBObjectStore;
+          if (database.objectStoreNames.contains(storeName)) {
+            store = database.transaction(storeName, 'readwrite').objectStore(storeName);
+          } else {
+            store = database.createObjectStore(storeName);
+          }
+          new Promise((resolve2, reject2) => {
+            store.transaction.oncomplete = resolve2;
+            store.transaction.onerror = reject2;
+            store.transaction.onabort = reject2;
+          }).then(() => {
+            resolve(store);
+          });
+        } catch (e) {
+          reject(e);
+        }
+      });
+    }
+  );
+
+  Cypress.Commands.overwrite('as', (originalAs, subject: any | IDBObjectStore, alias) => {
+    if (subject?.constructor?.name === 'IDBObjectStore') {
+      STORES.set(alias, subject);
+      return subject;
+    } else if (subject?.constructor?.name === 'IDBDatabase') {
+      DATABASES.set(alias, subject);
+      return subject;
+    } else {
+      return originalAs(alias);
```

Comment:
don't we need to pass the `subject` to `originalAs` as well?
```suggestion
      return originalAs(subject, alias);
```

Reply:
That was an awesome catch, thank you!